rfc/rfc-0003-developer-utils.md

```diff
@@ -0,0 +1,81 @@
+# RFC-#: A pod with utilities for better developer experience
+
+**Authors:** Swapnil Bawaskar
+
+**Status:**
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1362
+
+**Superseded by:** N/A
+
+**Supersedes:** https://github.com/projectriff/riff/pull/1359
+
+**Related:**
+
+
+## Problem
+The function developer would like to invoke the function for which they just created the `core deployer` or the `knative deployer`.
+When the streaming runtime is installed, they would like to test the function by getting events into the stream and look into the contents of the stream. There is no one step mechanism for accomplishing any of these currently.
+An [earlier proposal](https://github.com/projectriff/riff/pull/1359) was to provide some of this functionality in the riff cli, this proposal expands on Mark's comment https://github.com/projectriff/riff/pull/1359#discussion_r348617981
+
+### Anti-Goals
+We will only address this problem for development/demos, not production, so topics like auth/authz are out of scope for this document.
+
+## Solution
+We will ask the developers to run a `riff-developer-utils` pod in their development cluster. This pod will bundle a few binaries/scripts that users can invoke using `kubectl exec`. Since the commands will be run in-cluster, users won't need to run `kubectl port-forward`.
+We will have the following commands to start with:
+1. **invoke-core:** To invoke the given core deployer.  
+The command takes the form:  
+    ```
+    invoke-core <deployer-name> -n <namespace> <curl-params>
```

Comment:
how is the deployer name resolved to a host name? A vanilla pod won't have permission

Reply:
```suggestion
    invoke-core <deployer-name> -n <namespace> -- <curl-params>
```

Reply:
When running these commands using `kubectl exec` the separator `--` is already used. How about `-c` for curl options?

Reply:
updated proposal to use `-c`

Reply:
> how is the deployer name resolved to a host name? A vanilla pod won't have permission

this question wasn't addressed

Reply:
In my prototype, I ended up creating a clusterrole (with read permissions for riff resources) and clusterrolebinding for the service account that our pod ran under. Do you know if there is anything else we can do?